src/main/java/com/cryptomanager/repositories/PortfolioRepository.java

```diff
@@ -25,10 +25,13 @@ public void addPortfolio(Portfolio portfolio) {
 
         // Atualiza ou adiciona investimentos
         for (Investment investment : portfolio.getInvestments()) {
-            Portfolio existingPortfolio = existingPortfolios.stream()
-                    .filter(p -> p.getId().equals(portfolio.getId()))
-                    .findFirst()
-                    .orElse(null);
+            Portfolio existingPortfolio = null;
+            for (Portfolio p : existingPortfolios) {
+                if (p.getId().equals(portfolio.getId())) {
+                    existingPortfolio = p;
+                    break;
+                }
+            }
```

Comment:
Esse loop fica fora do loop de cima, nao faz sentido a cada investimento que for ser analisado procurar o mesmo portfolio repetidamente